File: T7/Ejercicios Clase/Ejercicio 2 T6/src/model/Trabajador.java
package model;

public abstract class Trabajador {

    private String nombre, apellido, dni;


    public Trabajador(){

    }

    public Trabajador(String nombre, String apellido, String dni) {
        this.nombre = nombre;
        this.apellido = apellido;
        this.dni = dni;
    }


    public void mostrarDatos(){
        System.out.println("Nombre: " + nombre + " Apellido: " + apellido + " DNI: " + dni);
    }


    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getApellido() {
        return apellido;
    }

    public void setApellido(String apellido) {
        this.apellido = apellido;
    }

    public String getDni() {
        return dni;
    }

    public void setDni(String dni) {
        this.dni = dni;
    }

}

File: Curso 2/PMPD/TrabajadoresHerencia/src/model/Trabajador.kt
package model

abstract class Trabajador(
    nombre: String,
    apellido: String,
    dni: String,
    var salario: Double,
    telefono: Int? = null,
    correo: String? = null,
    var seguro: Boolean? = false // el boolean ya es false por defecto asi que no hay que igualarlo a false
) : Persona(nombre, apellido, dni, telefono, correo) {

    // telefono, correo, SEGURO -> Optativo

    /*
    constructor(nombre: String,apellido: String,dni: String,salario: Int,seguro: Boolean) : this (nombre,apellido,dni,salario) {
        this.seguro = seguro
        this.telefono = telefono
        this.correo = correo
    }
     */

    // nombre, apellido, dni, SALARIO -> Obligatorios
    override fun mostrarDatos() {
        super.mostrarDatos()
        println("Salario: $salario")
        println("Seguro: $seguro")
    }

      abstract fun calcularSalarioNeto() : Double;

}